Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.cpp

```diff
@@ -198,8 +201,7 @@ void WebKitMediaKeySession::addKeyTimerFired()
             keyaddedEvent->setTarget(this);
             queueTaskToDispatchEvent(*this, TaskSource::Networking, WTFMove(keyaddedEvent));
 
-            ASSERT(m_keys);
-            m_keys->keyAdded();
+            Ref { *m_keys }->keyAdded();
```

Comment:
Perhaps we should keep the assert so we crash safely in the event it's nullptr?

Reply:
Crashing on nullptr is a prerequisite for Safer C++ and we already do that.

Reply:
> Crashing on nullptr is a prerequisite for Safer C++ and we already do that.

Uh, I do not think this is generally true. Dereferencing nullptr is undefined behavior and we have not solved this problem yet (because it regresses perf). Some smart pointers will RELEASE_ASSERT when referencing though but you should check, not assume. Here m_keys is a raw pointer so no check when doing `*` on it. Then `Ref`'s `operator->()` only has a debug assert, not a release assert.

Given the debug ASSERT in Ref, dropping the assert outside may be fine but I am not 100% sure. The compiler is free to optimize things and assume non-null when you pass a reference and not a pointer I believe. Ideally we want a RELEASE_ASSERT for safety though.